### ISSUE TITLE
Add centroids for all country codes at common breakpoints (500km, 200km, 100km, 50km?)

### ISSUE THREAD

**Issue by CloCkWeRX** [alltheplaces/alltheplaces#9373]
See also #6044

- We have eu centroids, which can be filtered by country - but not all spiders support that. Potentially this should be fixed. Couple of PL spiders as recent examples (#9369, #7589)
- ZA would be helpful (#9372)

This is probably to do easy but a bit opaque.

Later on, optimised centroid files described in #6044 would be really neat

**Comment by CloCkWeRX** [comment]
searchable_points/earth_centroids_iseadgg.md apparently are the docs

**Comment by CloCkWeRX** [comment]
@davidhicks I don't... quite understand the pipeline here - looking at say, https://github.com/alltheplaces/alltheplaces/blob/master/locations/searchable_points/de_centroids_iseadgg_50km_radius.meta I can understand there's a shape file used to clip an area, but I'm not really clear where it comes from.

If we were to do this for every country in the world; whats the current process?

https://help.openstreetmap.org/questions/53155/how-to-export-all-countries-boundary-from-osm-to-either-shape-file-or-kml-format ?
https://download.geofabrik.de/ ?

**Comment by davidhicks** [comment]
> @davidhicks I don't... quite understand the pipeline here - looking at say, https://github.com/alltheplaces/alltheplaces/blob/master/locations/searchable_points/de_centroids_iseadgg_50km_radius.meta I can understand there's a shape file used to clip an area, but I'm not really clear where it comes from.

The documentation lacks in this area a bit. It's documented instead in each `.meta` file see for example `us_centroids_iseadgg_458km_radius.meta` which states:

```
# us.shp generation method:
# 1. Download and extract: https://www.naturalearthdata.com/http//www.naturalearthdata.com/download/10m/cultural/ne_10m_admin_0_countries_lakes.zip
# 2. Execute: ogr2ogr -sql "SELECT OGR_GEOMETRY FROM ne_10m_admin_0_countries_lakes WHERE ISO_A2='US'" us.shp ne_10m_admin_0_countries_lakes.shp
```

**Comment by davidhicks** [comment]
I agree with this request. It was on my TODO list to generate these ISEA DGG centroid files for common radii for each country of the world, and commit them all at once. The script to regenerate them would also be provided to allow people to regenerate the centroid files periodically and whenever someone feels like doing it.

**Comment by davidhicks** [comment]
Links to naturalearthdata clipping shapes at https://www.naturalearthdata.com/downloads/10m-cultural-vectors/

The current link is `https://www.naturalearthdata.com/http//www.naturalearthdata.com/download/10m/cultural/ne_10m_admin_0_countries_lakes.zip` but I think they do referrer checking or something, as directly downloading from the link doesn't work. The script to automatically regenerate ISEA DGG centroid files will likely need to figure this out (e.g. supply the required Referer header, or generate and provide the required cookie, etc).

**Comment by CloCkWeRX** [comment]
Well, stuck at 
ubuntu@codespaces-88a392:/workspaces/DGGRID$ make install
make: *** No rule to make target 'install'.  Stop.

**Comment by CloCkWeRX** [comment]
Okay, they lost their make install and defined it in a dockerfile.

**Comment by davidhicks** [comment]
Fixed in #9685
